### PR TITLE
fix(sign-txn): Solve ios app launch issue

### PR DESCRIPTION
**Pull request by yigiterdev**
### Description
- On your first interaction (sign act) with the dApp browser asks you to "allow popups", if you allow it then it's all good if you don't allow it we'll show the redirect popup.
- There are some cases where Pera Wallet launches successfully but the redirect popup also shows up. 
- In some cases, safari can not ask permisson or redirect user to app. 

### Fixes
- We try to redirect users to app when they try to sign txn. In mobile safari, browser ask permission in every redirection try. But in other browsers, we successfully redirect user to mobile app. Check out the videos below. If we successfully redirect user to app, we will not show redirect modal. Other than that, we will show redirect modal (For example in safari).

  - Safari ->  https://user-images.githubusercontent.com/54077855/179740481-a0b0e610-3104-46f1-81f2-c1956f2f6db8.MP4
  - Other Browsers(Opera) -> https://user-images.githubusercontent.com/54077855/179728471-c8076ea9-a71f-4fe3-85d2-3f22a1473868.MP4
  
- We solved the 3rd problem by changing the position of a condition

### CODE REVIEW

File: src/PeraWalletConnect.ts [hunk 1]
```diff
@@ -186,6 +187,24 @@ class PeraWalletConnect {
       throw new Error("PeraWalletConnect was not initialized correctly.");
     }
 
+    if (isMobile()) {
+      let windowObj;
+
+      try {
+        // This is to automatically open the wallet app when trying to sign with it.
+        windowObj = window.open(PERA_WALLET_APP_DEEP_LINK, "_blank");
```
Comment (yigiterdev) [inline src/PeraWalletConnect.ts:195]: BTW, i'm not sure about naming. I'm open for suggestions.

File: src/PeraWalletConnect.ts [hunk 2]
```diff
@@ -186,6 +187,24 @@ class PeraWalletConnect {
       throw new Error("PeraWalletConnect was not initialized correctly.");
     }
 
+    if (isMobile()) {
+      let windowObj;
```
Comment (mucahit) [inline src/PeraWalletConnect.ts:191]:
```suggestion
      let peraWalletAppDeeplink;
```

File: src/PeraWalletConnect.ts (hunk 1 above)
Comment (mucahit) [inline src/PeraWalletConnect.ts:195]: Let's clean up inside of the `PeraWalletRedirectModal` component since we've moved the `window.open` stuff here.

**Review by mucahit — COMMENTED** [review]
thanks @yigiterdev this looks great

File: src/PeraWalletConnect.ts [hunk 3]
```diff
@@ -186,6 +187,24 @@ class PeraWalletConnect {
       throw new Error("PeraWalletConnect was not initialized correctly.");
     }
 
+    if (isMobile()) {
+      let peraWalletAppDeeplink;
+
+      try {
+        // This is to automatically open the wallet app when trying to sign with it.
+        peraWalletAppDeeplink = window.open(PERA_WALLET_APP_DEEP_LINK, "_blank");
```
Comment (mucahit) [inline src/PeraWalletConnect.ts:195]: Let's use `generatePeraWalletAppDeepLink()` utility

```suggestion
        peraWalletAppDeeplink = window.open(generatePeraWalletAppDeepLink(), "_blank");
```

**Review by mucahit — CHANGES_REQUESTED** [review]

**Review by mucahit — APPROVED** [review]

**Pull request merged**